hal/src/rtl872x/usart_hal.cpp

```diff
@@ -344,7 +346,7 @@ class Usart {
             }
             {
                 AtomicBlock atomic(this);
-                if (!isEnabled() || txBuffer_.empty()) {
+                if ((!isEnabled() || txBuffer_.empty()) && !busy_) {
```

Comment:
This should probably be checked above along with `transmitting_`? `while (transmitting_ || busy_)`

Reply:
Might try using `UART_WaitBusy()`, instead of introducing a new flag and configuring interrupt. This API is used in SDK exmaples.

Reply:
I was thinking the `while(true)` would keep it in the loop, but I suppose adding `while(transmitting_ || busy_){}` would prevent the atomic block from being set too frequently while `busy_ == true`.  I'll add that 👍🏼